esx_service/utils/kvESX.py

```diff
@@ -42,130 +47,156 @@
 # Default kv side car alignment
 KV_ALIGN = 4096
 
+# Flag to track the version of Python on the platform
+is_64bits = False
+
 # Load the disk lib API library
-def loadDiskLib():
+def loadDiskLib(lib_name):
     global lib
 
     if not lib:
-        lib = CDLL(DISK_LIB)
-        lib.DiskLib_Init.argtypes = []
-        lib.DiskLib_Init.restype = c_bool
-        lib.DiskLib_Init()
+       lib = CDLL(lib_name)
+       lib.DiskLib_Init.argtypes = []
+       lib.DiskLib_Init.restype = c_bool
+       lib.DiskLib_Init()
 
 
 # Loads the ESX library used to access proprietary ESX disk lib API.
 # Create arg/result definitions for those that we use.
-def kvESXInit():
-    global useSideCarCreate
 
-    # Load disklib APIs
-    loadDiskLib()
 
-    # Define all of the functions we are interested in
-    lib.DiskLib_OpenWithInfo.argtypes = [c_char_p, c_int32, POINTER(c_uint32),
-                                         POINTER(c_uint32), POINTER(c_uint32)]
-    lib.DiskLib_OpenWithInfo.restype = int
+    lib.DiskLib_SidecarMakeFileName.argtypes = [c_char_p, c_char_p]
+    lib.DiskLib_SidecarMakeFileName.restype = c_char_p
 
-    lib.DiskLib_Close.argtypes = [c_uint32]
-    lib.DiskLib_Close.restype = int
 
-    # Check if this library supports create API
-    try:
-        lib.DiskLib_SidecarCreate.argtypes = [c_uint32, c_char_p, c_uint64,
-                                              c_int32, POINTER(c_uint32)]
-        lib.DiskLib_SidecarCreate.restype = int
-        useSideCarCreate = True
-    except:
-        # do nothing
-        logging.debug(
-            "ESX version doesn't support create API, using open instead.")
+    return
 
-    lib.DiskLib_SidecarOpen.argtypes = [c_uint32, c_char_p, c_int32,
-                                        POINTER(c_uint32)]
-    lib.DiskLib_SidecarOpen.restype = int
+# Initialize 32-bit disk lib interfaces
+def kv_esx_init():
+    global is_64bits
+    global useSideCarCreate
 
-    lib.DiskLib_SidecarClose.argtypes = [c_uint32, c_char_p, POINTER(c_uint32)]
-    lib.DiskLib_SidecarClose.restype = int
+    # Define arg types for disk lib apis.
+    if sys.hexversion >= PYTHON64_VERSION:
```

Comment:
I am not sure why did you take all disklib init stuff from loadDiskLib() and moved to kv_esx_init() ? 
If it was to address my comment about spread of 32 vs 64 bit code, I think it's fine in init functions (which is also what I said). 

I think disklib-related init correct place in in loadDiskLib (or disklib_init(), using the proper convention). The current arrangement feels wrong. E..g. if I want to use DiskLib_OpenWithInfo, I for some reason need to call kv_esx_init() first, instead of just disklib_init().